jmh-core/src/main/java/org/openjdk/jmh/profile/XCTraceAsmProfiler.java

```diff
@@ -0,0 +1,279 @@
+            temporaryDirectory = XCTraceSupport.createTemporaryDirectoryName();
+        } catch (OptionException e) {
+            throw new ProfilerException(e.getMessage());
+        }
+    }
+
+    @Override
+    protected void addMyOptions(OptionParser parser) {
+        templateOpt = parser.accepts("template",
+                        "Path to or name of an Instruments template. " +
+                                "Use `xctrace list templates` to view available templates.")
+                .withRequiredArg()
+                .describedAs("string")
+                .ofType(String.class)
+                .defaultsTo("Time Profiler");
+        correctOpt = parser.accepts("fixStartTime", "Fix the start time by the time it took to launch.")
+                .withRequiredArg()
+                .describedAs("bool")
+                .ofType(Boolean.class)
+                .defaultsTo(true);
+    }
+
+    @Override
+    protected void parseEvents() {
+        Path profile = XCTraceSupport.findTraceFile(temporaryDirectory);
+        XCTraceSupport.exportTableOfContents(xctracePath, profile.toAbsolutePath().toString(),
+                perfParsedData.getAbsolutePath());
+        XCTraceTableOfContentsHandler handler = new XCTraceTableOfContentsHandler();
+        handler.parse(perfParsedData.file());
+        recordStartMs = handler.getRecordStartMs();
+        List<XCTraceTableHandler.XCTraceTableDesc> tables = handler.getSupportedTables();
+        if (tables.isEmpty()) {
+            throw new IllegalStateException("Profiling results does not contain tables supported by this profiler.");
+        }
+        if (tables.size() != 1) {
+            String supportedSchemas = Stream.of(XCTraceTableHandler.ProfilingTableType.values())
+                    .map(t -> t.tableName)
+                    .sorted()
+                    .collect(Collectors.joining(", "));
+            throw new IllegalStateException("There are multiple supported tables in output, " +
+                    "please make sure a template you chose produces only a single table with one of the following " +
+                    "schemas: " + supportedSchemas);
+        }
+        resultsTable = tables.get(0).getTableType();
+        XCTraceSupport.exportTable(xctracePath, profile.toAbsolutePath().toString(), perfParsedData.getAbsolutePath(),
+                resultsTable);
+    }
+
+    @Override
+    public void beforeTrial(BenchmarkParams params) {
+        super.beforeTrial(params);
+        temporaryDirectory.toFile().mkdirs();
+    }
+
+    @Override
+    public Collection<? extends Result> afterTrial(BenchmarkResult br, long pid, File stdOut, File stdErr) {
+        try {
+            BenchmarkResultMetaData md = br.getMetadata();
+            if (md != null) {
+                forkStartTimeMs = md.getStartTime();
+            }
+            return super.afterTrial(br, pid, stdOut, stdErr);
+        } finally {
+            XCTraceSupport.removeDirectory(temporaryDirectory);
```

Comment:
Do you really need `finally` here? Can we just record the result from super and then do `removeDirectory`?

Reply:
The intention was to remove the directory no matter what. 

`TempFile`s (like `hsLog`) should be removed by an on-exit hook, thanks to `deleteOnExit` called from `TempFileManager::create`. For directories, there's no such mechanism, so I decided to rely on `try-finally` instead.

Reply:
I was just asking if you want to avoid `finally` control flow here and just call it after `super.afterTrial` call. But that's optional, we can clean it up later.

Reply:
In that case, if `super.afterTrial` will throw something, the directory won't be deleted.

Reply:
Okay, fine.